From 92eafaf0f5902916ca87b8e3bd63040602ff18bf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 8 Jun 2015 13:59:25 -0700
Subject: [PATCH] linux-oe-g++: Invert conditional for defining QT_SOCKLEN_T

This helps to make sure that QT_SOCKLEN_T is defined to be 'int'
only when its glibc < 2 and not also for the libraries which may define
it as per standards but are not glibc, e.g. musl

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 mkspecs/linux-oe-g++/qplatformdefs.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mkspecs/linux-oe-g++/qplatformdefs.h b/mkspecs/linux-oe-g++/qplatformdefs.h
index dd12003..8623651 100644
--- a/mkspecs/linux-oe-g++/qplatformdefs.h
+++ b/mkspecs/linux-oe-g++/qplatformdefs.h
@@ -86,10 +86,10 @@
 
 #undef QT_SOCKLEN_T
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define QT_SOCKLEN_T            socklen_t
-#else
+#if defined(__GLIBC__) && (__GLIBC__ < 2)
 #define QT_SOCKLEN_T            int
+#else
+#define QT_SOCKLEN_T            socklen_t
 #endif
 
 #if defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE >= 500)
